### ISSUE TITLE
Non-required select

### ISSUE THREAD

**Issue by snoosmoomrik** [json-schema-form/angular-schema-form#708]
In this schema I defined 2 select boxes:
One of them is required, the other isn't.
```
  $scope.schema = {
    "type": "object",
    "required": ["list_required"],
    "properties": {
      "list_required": {
        "type": "string",
        "enum": ["1", "2", "3"]
      },
      "list_nonrequired": {
        "type": "string",
        "enum": ["1", "2", "3"]
      }
    }
  };
```
I expected that after selecting option in second (non-required) select box the empty option will remain to select it again, but it disappeared.

[Plunker demo](https://plnkr.co/edit/b3xwhNXliUkvIbUtHemw?p=preview)

To achieve needed behavior I usually used something like this:
```
<select ng-options="option as option.name for option in options">
    <option value=""></option>
</select>
```
But when I tried to override select.html with this solution, form validation returned error for empty value because of invalid type.

**Comment by Anthropic** [comment]
@snoosmoomrik you can add the extra option in the form definition, *should* achieve the same outcome.

**Comment by Anthropic** [comment]
Try this:
```javascript
$scope.form = [
  "list_required", {
    "key":"list_nonrequired",
    "type": "select",
    "titleMap": [
      {name:"Select", value: undefined},
      {name:"1", value:"1"},
      {name:"2", value:"2"},
      {name:"3", value:"3"}
    ]
  },
  {
      type: "submit",
      title: "Log"
  }
];
```

**Comment by snoosmoomrik** [comment]
Thank you! It's works now!

**Comment by vincetastic** [comment]
Adding the undefined value to the titleMap renders correctly, but leads to unexpected behavior when validating the form. Namely, this error:

"Invalid type: undefined (expected string)"

If I instead use a value of "", then it also renders correctly and avoids that error, however, when I actually select that value from the drop-down, a subsequent 'schemaFormRedraw' broadcast causes angular-schema-form to add a duplicate empty value to the drop-down.

In general, there seems to be some magic around when/why angular-schema-form adds/removes the empty choice to a select, and it's not quite clear how best to accommodate it.

**Comment by Anthropic** [comment]
@vincetastic you can also use null as an optional field type in your schema and use null as the default.

**Comment by vincetastic** [comment]
@Anthropic Thanks. Our use case is titleMaps that dynamically change in response to other selections. I played more with the undefined and it *seems* to work if I'm careful to only add the undefined entry to the titleMap if the titleMap is non-empty. I also had to update the corresponding model value to undefined when the list of options is otherwise empty.

For future reference, did you mean something like this:

    "mystringfield":{
        "oneOf": [
            {"type": "null"},
            {"type":"string"}
        ]
    }

It's a little wonky because it means I'll have to update schema properties simply to allow the empty selection, but I will definitely keep it in mind if I see unexpected behavior in the current design.
